pkg/controller/config-controller.go

```diff
@@ -211,6 +216,62 @@ func (r *CDIConfigReconciler) reconcileDefaultPodResourceRequirements(config *cd
 	return nil
 }
 
+func (r *CDIConfigReconciler) reconcileFilesystemOverhead(config *cdiv1.CDIConfig) error {
+	var globalOverhead cdiv1.Percent = "0.055"
+	var perStorageConfig = make(map[string]cdiv1.Percent)
+
+	log := r.log.WithName("CDIconfig").WithName("FilesystemOverhead")
+
+	// Avoid nil maps and segfaults for the initial case, where filesystemOverhead
+	// is nil for both the spec and the status.
+	if config.Status.FilesystemOverhead == nil {
+		log.Info("No filesystem overhead found in status, initializing to defaults")
+		config.Status.FilesystemOverhead = &cdiv1.FilesystemOverhead{
+			Global:       globalOverhead,
+			StorageClass: make(map[string]cdiv1.Percent),
+		}
+	}
+
+	if config.Spec.FilesystemOverhead != nil {
+		if valid, _ := validOverhead(config.Spec.FilesystemOverhead.Global); valid {
+			globalOverhead = config.Spec.FilesystemOverhead.Global
+		}
+		if config.Spec.FilesystemOverhead.StorageClass != nil {
+			perStorageConfig = config.Spec.FilesystemOverhead.StorageClass
+		}
+	}
+
+	// Set status global overhead
+	config.Status.FilesystemOverhead.Global = globalOverhead
+
+	// Set status per-storageClass overhead
+	storageClassList := &storagev1.StorageClassList{}
+	if err := r.client.List(context.TODO(), storageClassList, &client.ListOptions{}); err != nil {
+		return err
+	}
+
+	for _, storageClass := range storageClassList.Items {
+		storageClassName := storageClass.GetName()
+		storageClassNameOverhead, found := perStorageConfig[storageClassName]
+
+		if found {
+			valid, err := validOverhead(storageClassNameOverhead)
```

Comment:
can validation be enforced by CRD schema?

Reply:
There is a regexp on the field now, to enforce 0 <= overhead <= 1

Reply:
btw: I only managed to add it to the global overhead one, not to the map[storageclass]overhead.